docs/apps/components/data-grid.md

```diff
@@ -0,0 +1,208 @@
+|**Enable Range Selection**|Enable to select multiple cells in the data-grid and it provides you with minimum value, maximum value, average, sum, count|<img src= "/apps/components/img/datagrid_rangeselection.png">|
+|**Suppress Context Menu**|When disabled, it enables the mouse right-click which has options like export into `.CSV` file or `Excel`, copy or copy with headers etc|<img src= "/apps/components/img/datagrid_suppress.png">|
+|**Context Menu**|You can select from various options to add to your context menu|<img src= "/apps/components/img/datagrid_contextmenu.png">|
+|**Attribute Action**|Enable **Data Binding** to connect the data to UI. <br> Enable **Hidden** to action to hide the field|<img src= "/apps/components/img/datagrid_attributeaction.png">|
+
+We're talking about **Column Config** separately as it's a comprehensive component. Click on `+` to add a new column.
+
+1. **Basic**
+      * **Field**: Write the Field name.
+      * **Title**: Give the Title to your column.
+      * **Hide**: Toggle this option to either hide or show the column.
+      * **Sortable**: Enable to sort the columns
+      * **Checkbox Selection**: Allow users to choose multiple choices from a list.
+      * **Cell Data Type**: Choose data type from various options like Auto, Text, Number, Boolean, Date, Date String.
+      * **Auto Height**: Automatically adjusts the height of the column.
+      * **Wrap Text**: Automatically adjusts the arrangement of text to fit within a specified area.
+      * **Lock Visible**: It keeps key columns or rows always in view, even when scrolling, for focused data navigation and analysis.
+      * **Lock Position**: Set the Lock Position to either **Left**, **Right** or **None**.
+      * **Suppress Movable**: It locks the grid's layout, preventing manual reordering and ensuring consistent data presentation.
+
+<img src= "/apps/components/img/datagrid_basic.png" width="400">
+
+2.**Width**
+
+   * **Width**: Enter the width of the column in pixels.
+
+   * **Min Width**: Enter the minimum value width of the column in pixels.
+  
+   * **Max Width**: Enter the maximum value width of the column in pixels.
+  
+   * **Re-sizable**: Enable to resize the column width.
+
+<img src= "/apps/components/img/datagrid_width.png" width="400">
+
+3.**Value Format**
+Value format in a data-grid is a feature that controls how raw data values are displayed within individual cells.
+
+It allows you to customize the appearance of data to improve readability, highlight specific information, and ensure consistency across the grid.
+
+<img src= "/apps/components/img/datagrid_valueformat.png" width="400">
+
+4.**Display format**
+   
+   * **Format Type**: You can select the format options for a column from Date, Currency, Duration, Percent from Decimal, Percent from Number, Decimal, Second, Bytes.
+   
+   * **Link**: which is used to link from one page to another.
```

Comment:
This property is used to turn the values in the column into links that emit the onLinkClicked event

Reply:
done